### PR TITLE
fix(place type form) refactor 

### PR DESCRIPTION
**Pull request by sandymash**
- small refactor to place type select to solve bugs [#1314](https://dev.azure.com/spectrumFactory/CoronaI/_workitems/edit/1314), [#1338](https://dev.azure.com/spectrumFactory/CoronaI/_workitems/edit/1338), [#1339](https://dev.azure.com/spectrumFactory/CoronaI/_workitems/edit/1339), [#1341](https://dev.azure.com/spectrumFactory/CoronaI/_workitems/edit/1341)
- fixed sync between place type and sbubtype selection
- exported subplaces to redux to prevent refetching

### CODE REVIEW

File: client/src/commons/Forms/PlacesTypesAndSubTypes/usePlacesTypesAndSubTypes.ts [hunk 1]
```diff
@@ -7,15 +8,23 @@ import PlaceSubType from 'models/PlaceSubType';
...
-const usePlacesTypesAndSubTypes = (parameters: usePlacesTypesAndSubTypesIncome) => {
+const usePlacesTypesAndSubTypes = () => {
+    const placesSubTypesByTypes = useSelector<StoreStateType,PlacesSubTypesByTypes>(state => state.placeSubTypesByTypes);
+
 
-    const { setPlacesSubTypesByTypes } = parameters;
+    const subtypesFetched = useMemo(() => Object.keys(placesSubTypesByTypes).length > 0, [placesSubTypesByTypes]);
 
     const getPlacesSubTypesByTypes = () => {
+        if (Object.keys(placesSubTypesByTypes).length > 0) {
```
Comment (daniel7764) [inline client/src/commons/Forms/PlacesTypesAndSubTypes/usePlacesTypesAndSubTypes.ts:24]: Maybe use Boolean(Object.keys(placesSubTypesByTypes)) and if it's false then do the rest of the stuff and not do return;

Comment (sandymash) [reply client/src/commons/Forms/PlacesTypesAndSubTypes/usePlacesTypesAndSubTypes.ts:24]: im not sure i understand...

Comment (sandymash) [reply client/src/commons/Forms/PlacesTypesAndSubTypes/usePlacesTypesAndSubTypes.ts:24]: also Boolean(Object.keys(placesSubTypesByTypes)) will always be true

**Review by roymaniv — APPROVED** [review]

File: client/src/commons/Forms/PlacesTypesAndSubTypes/usePlacesTypesAndSubTypes.ts (hunk 1 above)
Comment (daniel7764) [reply client/src/commons/Forms/PlacesTypesAndSubTypes/usePlacesTypesAndSubTypes.ts:24]: What i was trying to say is try checcking if there's length at all or length == 0, and not if there's length > 0 then return nothing
Maybe Boolean(Object.keys(placesSubTypesByTypes).length)

**Review by y-golde — APPROVED** [review]

**Pull request merged**